ledger/src/shred.rs

```diff
@@ -576,6 +578,37 @@ impl Shred {
             Self::ShredData(_) => Err(Error::InvalidShredType),
         }
     }
+
+    /// Returns true if the other shred has the same ShredId, i.e. (slot, index,
+    /// shred-type), but different payload.
+    /// Retransmitter's signature is ignored when comparing payloads.
+    pub fn is_shred_duplicate(&self, other: &Shred) -> bool {
```

Comment:
Probably need some unit tests for this function too.
But it can be done later if we want to release and fix testnet first.

Reply:
Got it, i'll add a unit test in a follow up PR https://github.com/anza-xyz/agave/issues/2697.
Will aim to get this backported for tomorrow's testnet release.